be/src/formats/parquet/file_reader.cpp

```diff
@@ -67,60 +67,69 @@ Status FileReader::init(HdfsScannerContext* ctx) {
 }
 
 Status FileReader::_parse_footer() {
-    // try with buffer on stack
-    uint8_t local_buf[FOOTER_BUFFER_SIZE];
-    uint8_t* footer_buf = local_buf;
-    // we may allocate on heap if local_buf is not large enough.
-    DeferOp deferop([&] {
-        if (footer_buf != local_buf) {
-            delete[] footer_buf;
-        }
-    });
+    std::vector<char> footer_buffer;
+    ASSIGN_OR_RETURN(uint32_t footer_read_size, _get_footer_read_size());
+    footer_buffer.resize(footer_read_size);
```

Comment:
I think you can use struct in `be/src/util/raw_container.h` to ignore unnecessary memset(0)

Reply:
Improve in next pr.